server/models/procedures/invoicing.sql

```diff
@@ -771,3 +771,79 @@ BEGIN
       HAVING balance <> 0
     ) AS zz ORDER BY zz.date;
 END $$
+
+
+-- This Procedure retrieve the balace(debit, credit) of each invoices during a specified period
+
+DROP PROCEDURE IF EXISTS `UnbalancedInvoicePayments`$$
+CREATE   PROCEDURE `UnbalancedInvoicePayments`(IN dateFrom DATE, IN dateTo DATE)
+BEGIN
+  DECLARE _uuid, _debtor_uuid, _tempRef BINARY(16);
+  DECLARE _invoiceID VARCHAR(100);
+  DECLARE _date DATE;
+
+  DECLARE done BOOLEAN;
+  DECLARE curs1 CURSOR FOR 
+    SELECT  i.uuid, i.debtor_uuid, i.date 
```

Comment:
Would it be possible to just create a temporary table to hold invoice uuids?  A JOIN might be faster than a LOOP since it can use indexes more efficiently.  Like this:
```sql
CREATE TEMPORARY TABLE `invoices_to_find` AS 
  SELECT invoice.uuid, debtor_uuid, dm.text AS reference
  FROM invoice JOIN document_map dm ON invoice.uuid = dm.uuid
  WHERE DATE(i.date) BETWEEN DATE(dateFrom) AND DATE(dateTo);
```

Then we can do something like this: (I've renamed `UUID` to `i_uuid`, to make things clearer)
```sql
INSERT INTO stage_invoice_balance
  SELECT i_uuid, SUM(invoices_payments.debit_equiv) AS debit, SUM(invoices_payments.credit_equiv) AS credit
  FROM (

    -- posting journal invoices
    SELECT p.record_uuid AS i_uuid, p.debit_equiv, p.credit_equiv
      FROM posting_journal p JOIN invoices_to_find iv ON p.record_uuid = iv.uuid
      WHERE iv.debtor_uuid = p.entity_uuid

    UNION ALL

    -- general ledger invoices
    SELECT p.record_uuid AS i_uuid, p.debit_equiv, p.credit_equiv
      FROM general_ledger p JOIN invoices_to_find iv ON p.record_uuid = iv.uuid
      WHERE iv.debtor_uuid = p.entity_uuid

    UNION ALL

    -- posting_journal payments
    SELECT p.reference_uuid AS i_uuid, p.debit_equiv, p.credit_equiv
      FROM posting_journal p JOIN invoices_to_find iv ON p.reference_uuid = iv.uuid

    UNION ALL

    -- general_ledger payments
    SELECT p.reference_uuid AS i_uuid, p.debit_equiv, p.credit_equiv
      FROM general_ledger p JOIN invoices_to_find iv ON p.reference_uuid = iv.uuid
  )z
  GROUP BY i_uuid;
```